src/Elastic.Apm.StartupHook.Loader/Elastic.Apm.StartupHook.Loader.csproj

```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
+      <!-- TODO make this netstandard 2.0 and ref System.Runtime.Loader -->

```

Comment:
I remember there was some thinking here behind choosing `netcoreapp2.2`. This was the lowest possible target that seemed working.

Documentation on the startuphook feature can be found here: https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md

It says: "This means that `StartupHook.dll` needs to be built against the same or lower version of .NET Core than the app." And talks about the Trusted Platform Assemblies (TPA) list.

Also, this is a .NET Core 3+ feature, so I'm not sure netstandard 2.0 will be enough.

So I'm not totally sure if this would be possible.

Reply:
Yeah `AssemblyLoadContext` is not available in `netstandard2.0` but might be in `netstandard2.1`. If not then keeping `netcoreapp2.2.` makes the most sense for now and maybe discuss bumping this too separately.